### PR TITLE
Migrate login_client to null-safety

### LINKED ISSUE #13: Remove deprecated credentialsChangedCallback parameter
#12 deprecates the `credentialsChangedCallback` `LoginClient` constructor parameter.

Remove it before the next major version (v2.0.0).

### PR DESCRIPTION
**Pull request by Albert221**
Resolves #13.

### CODE REVIEW

**Comment by ResoDev** [comment]
LGTM, someone please make a review 😅

**Review by mchudy — APPROVED** [review]

**Comment by Albert221** [comment]
MAJOR needs to be bumped after the 5.0.0 release

**Review by shilangyu — APPROVED** [review]

**Review by RobertOdrowazLNCD — APPROVED** [review]

**Comment by Albert221** [comment]
@mateuszwojtczak waiting for your final approval

**Review by mateuszwojtczak — APPROVED** [review]

**Pull request merged**
